### DISCUSSION TITLE
Replication without zookeeper

### DISCUSSION THREAD

**Discussion by mrceperka** [ClickHouse/ClickHouse#29830] [Q&A] [answered]
Hello, 

I'm wondering if it's possible to setup **"replication" without zookeeper**. 
I think it's possible, but I'd like to know what you/others think.

My idea is simple. 
:bulb: Instead of using `zk`, let's use `nfs` for sharing whole `/var/clickhouse` setup over multiple servers.

I am thinking of 2 servers:
 - first server - all the mutations, ddl changes, etc. would happen here and only here.
 - second server - would by in read-only mode (hopefully the data in `/var/clickhouse` would not get corrupted)
 
 I am not sure if this setup could benefit from `Distributed` table engine, probably not because it's not a cluster :/
 
 What do you think of it? 
 Are there any better approaches without using `zk`? 
 Or is it better to "just" insert data to multiple servers  rather than messing with nfs?
 
 Thanks in advance.

**Comment by morozovsk** [comment]
nfs is not replication.
better to use [ClickHouse Keeper](https://clickhouse.com/docs/en/operations/clickhouse-keeper/).

**Comment by mrceperka** [comment]
Thanks! Is is prod ready already? Would be cool

**Reply by morozovsk** [reply]
:) please read at least first line:
>### pre-production

**Reply by mrceperka** [reply]
Yeah, thing is... I am setting up zk based cluster. Should I wait for clickhouse-keeper? Should I migrate from zk to keeper when keeper prod-ready?

I'd say that nfs is some kind of replication, but keeper or zk are hopefully better.

**Comment by mrceperka** [comment, marked as answer]
Well, I am gonna close this. For now, I will be using zookeeper.
